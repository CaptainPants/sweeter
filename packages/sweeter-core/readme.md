# Sweeter Core
This package contains the share core of Sweeter. 

This includes:
- Signals
- Base Components
- Base Hooks
- Context