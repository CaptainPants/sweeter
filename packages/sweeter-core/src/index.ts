import './jsx-globals.js';
import './SweeterExtensionPoints.js';

export * from './addExplicitStrongReference.js';
export * from './components/index.js';
export * from './context/index.js';
export * from './executionContext/index.js';
export * from './lazy/index.js';
export * from './middleware/index.js';
export * from './runtime/index.js';
export * from './signals/index.js';
export * from './types.js';
export * from './test/index.js';
export * from './utility/index.js';