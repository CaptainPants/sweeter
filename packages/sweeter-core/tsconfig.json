{
    "extends": "../../common-config/tsconfig.json",
    "include": [
        "src/**/*.ts",
        "src/**/*.tsx"
    ],
    "compilerOptions": {
        "rootDir": "src",
        "lib": [
            "ES2022", // No DOM here
        ],
        "noEmit": true,
        "paths":{
        }
    }
}
