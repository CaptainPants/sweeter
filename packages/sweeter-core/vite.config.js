import sweeterPlugin, { watchDependenciesPlugin } from '@captainpants/rollup-plugin-sweeter';
import { resolve } from 'path';
import nodeExternals from 'rollup-plugin-node-externals';
import { defineConfig } from 'vite';
import circularDependency from 'vite-plugin-circular-dependency';
import dts from 'vite-plugin-dts';

export default defineConfig({
    build: {
        lib: {
            // Could also be a dictionary or array of multiple entry points
            // eslint-disable-next-line no-undef
            entry: resolve(__dirname, './src/index.ts'),
            fileName: 'index',
            name: 'index',
            formats: ['es'],
        },
        outDir: 'dist',
        minify: false,
        rollupOptions: {
        },
        target: "ESNext",
    },
    test: {
        poolOptions: {
            forks: {
                execArgv: ["--expose-gc"]
            }
        },
    },
    plugins: [
        dts({ clearPureImport: false }), 
        nodeExternals(), 
        circularDependency({ circleImportThrowErr: true }), 
        sweeterPlugin({ projectName: '@captainpants/sweeter-core', roots: [__dirname] }),
        watchDependenciesPlugin({ dependencies: [{ namePattern: '@captainpants/*', filesPattern: 'dist/built.txt', buildCompleteNoticePath: 'dist/built.txt' }] })
    ]
});
