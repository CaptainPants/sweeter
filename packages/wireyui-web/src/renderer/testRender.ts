import { WebRenderer } from './WebRenderer.js';

export interface TestRenderResult {
    readonly nodes: NodeListOf<ChildNode>;
    readonly getHTML: () => string;
    readonly dispose: () => void;
}

/**
 * Convenience function for rendering isolated JSX chunks for unit tests
 * @param callback
 * @returns
 */
export function testRender(callback: () => JSX.Element): TestRenderResult {
    const renderer = new WebRenderer();

    const root = document.createElement('div');

    const dispose = renderer.start(root, callback);

    return {
        nodes: root.childNodes,
        getHTML: () => root.innerHTML,
        dispose,
    };
}
