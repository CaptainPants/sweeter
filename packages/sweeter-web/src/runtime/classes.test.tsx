/* @jsxImportSource ../. */

import { $mutable } from '@captainpants/sweeter-core';

import { testRender } from '../test/testRender.js';

it('class array with signals', () => {
    const cls1 = $mutable('alpha');
    const cls2 = $mutable('beta');

    const res = testRender(() => <div class={[cls1, cls2, 'gamma']}></div>);

    expect(res.getHTML()).toMatchSnapshot();

    cls2.value = 'delta';

    expect(res.getHTML()).toMatchSnapshot();

    res.dispose();
});

it('class object', () => {
    const on = $mutable(false);

    const res = testRender(() => <div class={[{ alpha: on }, 'beta']}></div>);

    expect(res.getHTML()).toMatchSnapshot();

    on.value = true;

    expect(res.getHTML()).toMatchSnapshot();

    res.dispose();
});
