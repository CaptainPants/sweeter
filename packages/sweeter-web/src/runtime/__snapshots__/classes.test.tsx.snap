// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`class array with signals 1`] = `"<div class="alpha beta gamma"></div>"`;

exports[`class array with signals 2`] = `"<div class="alpha delta gamma"></div>"`;

exports[`class object 1`] = `"<div class="beta"></div>"`;

exports[`class object 2`] = `"<div class="alpha beta"></div>"`;
